src/internal/account.ts

```diff
@@ -137,12 +141,45 @@ export async function getTransactions(args: {
   options?: PaginationArgs;
 }): Promise<TransactionResponse[]> {
   const { aptosConfig, accountAddress, options } = args;
-  return paginateWithCursor<{}, TransactionResponse[]>({
+  // TODO: Ideally indexer should provide one API that returns all transactions details in one go
+  // But now we have to query all transaction versions first and then query each transaction by version
+  const versions = await getAccountAllTransactionVersions({ aptosConfig, accountAddress, options });
+  const results = [];
+  for (const version of versions) {
+    const tx = getTransactionByVersion({ aptosConfig, ledgerVersion: version });
+    results.push(tx);
+  }
+  return Promise.all(results);
+}
```

Comment:
This significantly changes the behavior of this function from being a paginated API that is associated with an account, to having a non-paginated API across two different services.

If this is the case that this is important, it should do whatever query the explorer and wallets do, or have a dedicated query for it.

Additionally, this changes what is a 1 network call, to 1 + n network calls.

In a limited network environment, it would be slow or eat up all your bandwidth.